chains/evm/calls/contracts/deposit/deposit.go

```diff
@@ -1,25 +1,30 @@
 package deposit
 
 import (
-	"github.com/ethereum/go-ethereum/common/math"
 	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/math"
 )
 
-func ConstructErc20DepositData(destRecipient []byte, amount *big.Int) []byte {
+func ConstructErc20DepositData(destRecipient []byte, amount *big.Int, priority uint8) []byte {
 	var data []byte
-	data = append(data, math.PaddedBigBytes(amount, 32)...)
-	data = append(data, math.PaddedBigBytes(big.NewInt(int64(len(destRecipient))), 32)...)
-	data = append(data, destRecipient...)
+	data = append(data, math.PaddedBigBytes(amount, 32)...)                                  // Amount
+	data = append(data, math.PaddedBigBytes(big.NewInt(int64(len(destRecipient))), 32)...)   // length of recipient
+	data = append(data, destRecipient...)                                                    // Recipient
+	data = append(data, math.PaddedBigBytes(big.NewInt(int64(len([]byte{priority}))), 1)...) // Length of priority
+	data = append(data, []byte{priority}...)                                                 // Priority
 	return data
 }
 
-func ConstructErc721DepositData(destRecipient []byte, tokenId *big.Int, metadata []byte) []byte {
```

Comment:
Should we handle the priority data in ConstructGenericDepositData also? @P1sar

Reply:
I have answered this question in your PR actually

Reply:
Let's leave it as is for the time being ->
https://github.com/ChainSafe/chainbridge-core/pull/242/files#r794403715